client/v2/autocli/flag/builder.go

```diff
@@ -162,38 +162,42 @@ func (b *Builder) addMessageFlags(ctx *context.Context, flagSet *pflag.FlagSet,
 			messageBinder.hasOptional = true
 		}
 
-		field := fields.ByName(protoreflect.Name(arg.ProtoField))
-		if field == nil {
-			return nil, fmt.Errorf("can't find field %s on %s", arg.ProtoField, messageType.Descriptor().FullName())
+		if arg.FlattenFields {
```

Comment:
We actually don't want the user to have to precise this in their autocli options.
I think ideally it should just work.

if you put `permissions` you it set the whole struct. If you put `permissions.level` it it checks the inner field. I think splitting by `.` may do the job here right.
The big q is how many layer do we accept. Can you do `permissions.foo.bar` as well?

Reply:
ok. If you set only `permissions` should it flatten the whole struct or should expect a json?

In the first case, should we provide an option to specify a json file? it may be better ux if message is really big.

About the layer that's a good question. I'll try to find a recursive way so there's no limit to it.

Reply:
I meant, it should expect a json yeah, like today

Reply:
> In the first case, should we provide an option to specify a json file? it may be better ux if message is really big.

It can get messy if you have multiple input expecting a json, so to simplify things I'd say no.
If you want a better UX you break out the json in multiple inputs by using the feature added here